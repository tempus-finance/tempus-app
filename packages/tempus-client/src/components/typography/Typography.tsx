import { CSSProperties, FC } from 'react';

type TypographyVariant = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'body-text' | 'disclaimer-text';
type TypographyColor = 'default' | 'accent' | 'inverted';

const typographyStyleMap = new Map<TypographyVariant, CSSProperties>();
typographyStyleMap.set('h1', {
  fontFamily: "'Inter', sans-serif",
  fontWeight: 900,
  fontSize: '36px',
  fontStyle: 'normal',
  lineHeight: '44px',
});
typographyStyleMap.set('h2', {
  fontFamily: "'Inter', sans-serif",
  fontWeight: 900,
  fontSize: '26px',
  fontStyle: 'normal',
  lineHeight: '31px',
});
typographyStyleMap.set('h3', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 500,
  fontSize: '22px',
  fontStyle: 'normal',
  lineHeight: '26px',
});
typographyStyleMap.set('h4', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 700,
  fontSize: '18px',
  fontStyle: 'normal',
  lineHeight: '21px',
});
typographyStyleMap.set('h5', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 500,
  fontSize: '16px',
  fontStyle: 'normal',
  lineHeight: '19px',
});
typographyStyleMap.set('body-text', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 400,
  fontSize: '16px',
  fontStyle: 'normal',
  lineHeight: '19px',
});
typographyStyleMap.set('disclaimer-text', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 400,
  fontSize: '12px',
  fontStyle: 'normal',
  lineHeight: '14px',
});

interface TypographyProps {
  variant: TypographyVariant;
  color: TypographyColor;
}

const Typography: FC<TypographyProps> = props => {
  let color: string;
  switch (props.color) {
    case 'default':
      color = '#222222';
      break;
    case 'accent':
      color = '#F5AC37';
      break;
    case 'inverted':
      color = '#FFFFFF';
      break;
  }

  return (
    <div
      style={{
        ...typographyStyleMap.get(props.variant),
        color: color,
      }}
    >
      {props.children}
    </div>
  );
};
export default Typography;
