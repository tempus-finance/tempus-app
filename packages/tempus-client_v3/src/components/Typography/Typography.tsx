import React, { CSSProperties, FC } from 'react';
import parse from 'html-react-parser';

type TypographyVariant =
  | 'header'
  | 'subheader'
  | 'title'
  | 'subtitle'
  | 'body-primary'
  | 'body-secondary'
  | 'boyd-tertiary';

type TypographyColor = 'text-primary';

type TypographyWeight = 'regular' | 'medium' | 'bold';

type TypographySpacing = 'regular' | 'mono';

const typographyVariantMap = new Map<TypographyVariant, CSSProperties>();
typographyVariantMap.set('header', {
  fontStyle: 'normal',
  fontSize: '40px',
  lineHeight: '48px',
});
typographyVariantMap.set('subheader', {
  fontStyle: 'normal',
  fontSize: '32px',
  lineHeight: '40px',
});
typographyVariantMap.set('title', {
  fontStyle: 'normal',
  fontSize: '24px',
  lineHeight: '32px',
});
typographyVariantMap.set('subtitle', {
  fontStyle: 'normal',
  fontSize: '20px',
  lineHeight: '28px',
});
typographyVariantMap.set('body-primary', {
  fontStyle: 'normal',
  fontSize: '16px',
  lineHeight: '24px',
});
typographyVariantMap.set('body-secondary', {
  fontStyle: 'normal',
  fontSize: '12px',
  lineHeight: '16px',
});
typographyVariantMap.set('boyd-tertiary', {
  fontStyle: 'normal',
  fontSize: '10px',
  lineHeight: '16px',
});

const typographyColorMap = new Map<TypographyColor, string>();
typographyColorMap.set('text-primary', '#222222');

const typographyWeightMap = new Map<TypographyWeight, number>();
typographyWeightMap.set('regular', 400);
typographyWeightMap.set('medium', 500);
typographyWeightMap.set('bold', 700);

const typographySpacingMap = new Map<TypographySpacing, string>();
typographySpacingMap.set('regular', "'DM Sans', sans-serif");
typographySpacingMap.set('mono', "'Azeret Mono', monospace");

interface TypographyProps {
  variant: TypographyVariant;
  color?: TypographyColor;
  weight?: TypographyWeight;
  spacing?: TypographySpacing;
  html?: string;
}

const Typography: FC<TypographyProps & React.HTMLProps<HTMLDivElement>> = props => {
  const { variant, color, weight, spacing, html, children } = props;

  let fontColor: string | undefined;
  if (color) {
    fontColor = typographyColorMap.get(color);
  } else {
    fontColor = typographyColorMap.get('text-primary');
  }

  let fontWeight: number | undefined;
  if (weight) {
    fontWeight = typographyWeightMap.get(weight);
  } else {
    fontWeight = typographyWeightMap.get('regular');
  }

  let fontFamily: string | undefined;
  if (spacing) {
    fontFamily = typographySpacingMap.get(spacing);
  } else {
    fontFamily = typographySpacingMap.get('regular');
  }

  return (
    <div
      style={{
        ...typographyVariantMap.get(variant),
        color: fontColor,
        fontWeight: fontWeight,
        fontFamily: fontFamily,
      }}
    >
      {html ? parse(html) : children}
    </div>
  );
};
export default Typography;
