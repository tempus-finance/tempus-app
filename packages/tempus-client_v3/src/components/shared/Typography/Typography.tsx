import React, { CSSProperties, FC } from 'react';
import parse from 'html-react-parser';
import { colors } from '../Color';

type TypographyVariant =
  | 'header'
  | 'subheader'
  | 'title'
  | 'subtitle'
  | 'body-primary'
  | 'body-secondary'
  | 'body-tertiary';

export type TypographyColor =
  | 'text-primary'
  | 'text-inverted'
  | 'text-success'
  | 'text-disabled'
  | 'text-disabled-secondary';

export type TypographyWeight = 'regular' | 'medium' | 'bold';

type TypographyType = 'regular' | 'mono';

const typographyVariantMap = new Map<TypographyVariant, CSSProperties>();
typographyVariantMap.set('header', {
  fontStyle: 'normal',
  fontSize: '40px',
  lineHeight: '48px',
});
typographyVariantMap.set('subheader', {
  fontStyle: 'normal',
  fontSize: '32px',
  lineHeight: '40px',
});
typographyVariantMap.set('title', {
  fontStyle: 'normal',
  fontSize: '24px',
  lineHeight: '32px',
});
typographyVariantMap.set('subtitle', {
  fontStyle: 'normal',
  fontSize: '20px',
  lineHeight: '28px',
});
typographyVariantMap.set('body-primary', {
  fontStyle: 'normal',
  fontSize: '16px',
  lineHeight: '24px',
});
typographyVariantMap.set('body-secondary', {
  fontStyle: 'normal',
  fontSize: '12px',
  lineHeight: '16px',
});
typographyVariantMap.set('body-tertiary', {
  fontStyle: 'normal',
  fontSize: '10px',
  lineHeight: '16px',
});

const typographyColorMap = new Map<TypographyColor, string>();
typographyColorMap.set('text-primary', colors.textPrimary);
typographyColorMap.set('text-inverted', colors.textInverted);
typographyColorMap.set('text-success', colors.textSuccess);
typographyColorMap.set('text-disabled', colors.textDisabled);
typographyColorMap.set('text-disabled-secondary', colors.textDisabledSecondary);

const typographyWeightMap = new Map<TypographyWeight, number>();
typographyWeightMap.set('regular', 400);
typographyWeightMap.set('medium', 500);
typographyWeightMap.set('bold', 700);

const typographyTypeMap = new Map<TypographyType, string>();
typographyTypeMap.set('regular', "'DM Sans', sans-serif");
typographyTypeMap.set('mono', "'Azeret Mono', monospace");

interface TypographyProps {
  variant: TypographyVariant;
  color?: TypographyColor;
  opacity?: number;
  weight?: TypographyWeight;
  type?: TypographyType;
  html?: string;
}

const Typography: FC<TypographyProps & React.HTMLProps<HTMLDivElement>> = props => {
  const { variant, color, weight, opacity = 1, type, html, children } = props;

  let fontColor: string | undefined;
  if (color) {
    fontColor = typographyColorMap.get(color);
  } else {
    fontColor = typographyColorMap.get('text-primary');
  }

  let fontWeight: number | undefined;
  if (weight) {
    fontWeight = typographyWeightMap.get(weight);
  } else {
    fontWeight = typographyWeightMap.get('regular');
  }

  let fontFamily: string | undefined;
  if (type) {
    fontFamily = typographyTypeMap.get(type);
  } else {
    fontFamily = typographyTypeMap.get('regular');
  }

  return (
    <div
      style={{
        ...typographyVariantMap.get(variant),
        color: fontColor,
        fontWeight: fontWeight,
        fontFamily: fontFamily,
        opacity,
      }}
    >
      {html ? parse(html) : children}
    </div>
  );
};
export default Typography;
