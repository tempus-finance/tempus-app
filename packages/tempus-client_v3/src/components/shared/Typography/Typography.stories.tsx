import React, { CSSProperties } from 'react';

import Typography from './Typography';

export default {
  title: 'Typography',
  component: Typography,
  argTypes: {},
};

const style: CSSProperties = {
  background: 'rgba(0, 0, 0, 0.1)',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  padding: '10px',
};

export const Header = () => (
  <div style={style}>
    <Typography variant="header">Header</Typography>
    <br />
    <Typography variant="header" weight="medium">
      Header Medium
    </Typography>
    <br />
    <Typography variant="header" weight="bold">
      Header Bold
    </Typography>
    <br />
    <Typography variant="header" type="mono">
      Header Mono
    </Typography>
    <br />
    <Typography variant="header" weight="medium" type="mono">
      Header Mono Medium
    </Typography>
    <br />
    <Typography variant="header" weight="bold" type="mono">
      Header Mono Bold
    </Typography>
  </div>
);

export const Subheader = () => (
  <div style={style}>
    <Typography variant="subheader">Subheader</Typography>
    <br />
    <Typography variant="subheader" weight="medium">
      Subheader Medium
    </Typography>
    <br />
    <Typography variant="subheader" weight="bold">
      Subheader Bold
    </Typography>
    <br />
    <Typography variant="subheader" type="mono">
      Subheader Mono
    </Typography>
    <br />
    <Typography variant="subheader" weight="medium" type="mono">
      Subheader Mono Medium
    </Typography>
    <br />
    <Typography variant="subheader" weight="bold" type="mono">
      Subheader Mono Bold
    </Typography>
  </div>
);

export const Title = () => (
  <div style={style}>
    <Typography variant="title">Title</Typography>
    <br />
    <Typography variant="title" weight="medium">
      Title Medium
    </Typography>
    <br />
    <Typography variant="title" weight="bold">
      Title Bold
    </Typography>
    <br />
    <Typography variant="title" type="mono">
      Title Mono
    </Typography>
    <br />
    <Typography variant="title" weight="medium" type="mono">
      Title Mono Medium
    </Typography>
    <br />
    <Typography variant="title" weight="bold" type="mono">
      Title Mono Bold
    </Typography>
  </div>
);

export const Subtitle = () => (
  <div style={style}>
    <Typography variant="subtitle">Subtitle</Typography>
    <br />
    <Typography variant="subtitle" weight="medium">
      Subtitle Medium
    </Typography>
    <br />
    <Typography variant="subtitle" weight="bold">
      Subtitle Bold
    </Typography>
    <br />
    <Typography variant="subtitle" type="mono">
      Subtitle Mono
    </Typography>
    <br />
    <Typography variant="subtitle" weight="medium" type="mono">
      Subtitle Mono Medium
    </Typography>
    <br />
    <Typography variant="subtitle" weight="bold" type="mono">
      Subtitle Mono Bold
    </Typography>
  </div>
);

export const BodyPrimary = () => (
  <div style={style}>
    <Typography variant="body-primary">Body Primary</Typography>
    <br />
    <Typography variant="body-primary" weight="medium">
      Body Primary Medium
    </Typography>
    <br />
    <Typography variant="body-primary" weight="bold">
      Body Primary Bold
    </Typography>
    <br />
    <Typography variant="body-primary" type="mono">
      Body Primary Mono
    </Typography>
    <br />
    <Typography variant="body-primary" weight="medium" type="mono">
      Body Primary Mono Medium
    </Typography>
    <br />
    <Typography variant="body-primary" weight="bold" type="mono">
      Body Primary Mono Bold
    </Typography>
  </div>
);

export const BodySecondary = () => (
  <div style={style}>
    <Typography variant="body-secondary">Body Secondary</Typography>
    <br />
    <Typography variant="body-secondary" weight="medium">
      Body Secondary Medium
    </Typography>
    <br />
    <Typography variant="body-secondary" weight="bold">
      Body Secondary Bold
    </Typography>
    <br />
    <Typography variant="body-secondary" type="mono">
      Body Secondary Mono
    </Typography>
    <br />
    <Typography variant="body-secondary" weight="medium" type="mono">
      Body Secondary Mono Medium
    </Typography>
    <br />
    <Typography variant="body-secondary" weight="bold" type="mono">
      Body Secondary Mono Bold
    </Typography>
  </div>
);

export const BodyTertiary = () => (
  <div style={style}>
    <Typography variant="body-tertiary">Body Tertiary</Typography>
    <br />
    <Typography variant="body-tertiary" weight="medium">
      Body Tertiary Medium
    </Typography>
    <br />
    <Typography variant="body-tertiary" weight="bold">
      Body Tertiary Bold
    </Typography>
    <br />
    <Typography variant="body-tertiary" type="mono">
      Body Tertiary Mono
    </Typography>
    <br />
    <Typography variant="body-tertiary" weight="medium" type="mono">
      Body Tertiary Mono Medium
    </Typography>
    <br />
    <Typography variant="body-tertiary" weight="bold" type="mono">
      Body Tertiary Mono Bold
    </Typography>
  </div>
);
