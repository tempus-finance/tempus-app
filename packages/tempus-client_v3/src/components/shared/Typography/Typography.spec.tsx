import { render } from '@testing-library/react';
import Typography, {
  TypographyColor,
  TypographyProps,
  TypographyType,
  TypographyVariant,
  TypographyWeight,
} from './Typography';

const defaultProps: TypographyProps = {
  variant: 'title',
};

const subject = (props: TypographyProps) => render(<Typography {...props} />);

describe('Typography', () => {
  (
    [
      'page-navigation',
      'header',
      'subheader',
      'title',
      'subtitle',
      'body-primary',
      'body-secondary',
      'body-tertiary',
    ] as TypographyVariant[]
  ).forEach(variant => {
    it(`render a text with variant ${variant}`, () => {
      const props = { ...defaultProps, variant, children: 'simple text' };
      const { container, queryByText } = subject(props);
      const text = queryByText(props.children);

      expect(container).not.toBeNull();
      expect(text).not.toBeNull();

      expect(container).toMatchSnapshot();
    });

    it(`render html with variant ${variant}`, () => {
      const str = 'simple text';
      const props = { ...defaultProps, variant, html: `<span>${str}</span>` };
      const { container, queryByText } = subject(props);
      const text = queryByText(str);

      expect(container).not.toBeNull();
      expect(text).not.toBeNull();

      expect(container).toMatchSnapshot();
    });
  });

  (
    [
      'text-primary',
      'text-secondary',
      'primary-dark',
      'text-primary-inverted',
      'text-success',
      'text-disabled',
      'text-disabled-secondary',
      'text-caption',
      'text-error',
    ] as TypographyColor[]
  ).forEach(color => {
    it(`render a text with color ${color}`, () => {
      const props = { ...defaultProps, color, children: 'simple text' };
      const { container, queryByText } = subject(props);
      const text = queryByText(props.children);

      expect(container).not.toBeNull();
      expect(text).not.toBeNull();

      expect(container).toMatchSnapshot();
    });

    it(`render html with color ${color}`, () => {
      const str = 'simple text';
      const props = { ...defaultProps, color, html: `<span>${str}</span>` };
      const { container, queryByText } = subject(props);
      const text = queryByText(str);

      expect(container).not.toBeNull();
      expect(text).not.toBeNull();

      expect(container).toMatchSnapshot();
    });
  });

  (['regular', 'medium', 'bold'] as TypographyWeight[]).forEach(weight => {
    it(`render a text with weight ${weight}`, () => {
      const props = { ...defaultProps, weight, children: 'simple text' };
      const { container, queryByText } = subject(props);
      const text = queryByText(props.children);

      expect(container).not.toBeNull();
      expect(text).not.toBeNull();

      expect(container).toMatchSnapshot();
    });

    it(`render html with weight ${weight}`, () => {
      const str = 'simple text';
      const props = { ...defaultProps, weight, html: `<span>${str}</span>` };
      const { container, queryByText } = subject(props);
      const text = queryByText(str);

      expect(container).not.toBeNull();
      expect(text).not.toBeNull();

      expect(container).toMatchSnapshot();
    });
  });

  (['regular', 'mono'] as TypographyType[]).forEach(type => {
    it(`render a text with type ${type}`, () => {
      const props = { ...defaultProps, type, children: 'simple text' };
      const { container, queryByText } = subject(props);
      const text = queryByText(props.children);

      expect(container).not.toBeNull();
      expect(text).not.toBeNull();

      expect(container).toMatchSnapshot();
    });

    it(`render html with type ${type}`, () => {
      const str = 'simple text';
      const props = { ...defaultProps, type, html: `<span>${str}</span>` };
      const { container, queryByText } = subject(props);
      const text = queryByText(str);

      expect(container).not.toBeNull();
      expect(text).not.toBeNull();

      expect(container).toMatchSnapshot();
    });
  });
});
