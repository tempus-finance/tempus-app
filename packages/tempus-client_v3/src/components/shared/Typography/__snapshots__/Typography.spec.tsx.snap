// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography render a text with color primary-dark 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(62, 78, 110); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with color text-caption 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(245, 172, 55); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with color text-disabled 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(122, 122, 122); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with color text-disabled-secondary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(191, 192, 192); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with color text-error 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(255, 15, 15); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with color text-primary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with color text-primary-inverted 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(255, 255, 255); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with color text-secondary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(71, 67, 80); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with color text-success 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(75, 181, 67); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with type mono 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'Azeret Mono', monospace; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with type regular 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with variant body-primary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with variant body-secondary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with variant body-tertiary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 10px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with variant header 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 40px; line-height: 48px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with variant page-navigation 1`] = `
<div>
  <div
    style="font-style: bold; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with variant subheader 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 32px; line-height: 40px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with variant subtitle 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with variant title 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with weight bold 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with weight medium 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render a text with weight regular 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    simple text
  </div>
</div>
`;

exports[`Typography render html with color primary-dark 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(62, 78, 110); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with color text-caption 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(245, 172, 55); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with color text-disabled 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(122, 122, 122); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with color text-disabled-secondary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(191, 192, 192); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with color text-error 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(255, 15, 15); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with color text-primary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with color text-primary-inverted 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(255, 255, 255); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with color text-secondary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(71, 67, 80); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with color text-success 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(75, 181, 67); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with type mono 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'Azeret Mono', monospace; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with type regular 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with variant body-primary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with variant body-secondary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with variant body-tertiary 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 10px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with variant header 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 40px; line-height: 48px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with variant page-navigation 1`] = `
<div>
  <div
    style="font-style: bold; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with variant subheader 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 32px; line-height: 40px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with variant subtitle 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with variant title 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with weight bold 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with weight medium 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;

exports[`Typography render html with weight regular 1`] = `
<div>
  <div
    style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    <span>
      simple text
    </span>
  </div>
</div>
`;
